src/python/pants/task/task.py

```diff
@@ -333,14 +333,19 @@ def invalidated(self,
       self.fingerprint)
     cache_manager = InvalidationCacheManager(self.workdir,
                                              cache_key_generator,
-                                             self._build_invalidator_dir,
+                                             self._build_invalidator(),
                                              invalidate_dependents,
                                              fingerprint_strategy=fingerprint_strategy,
                                              invalidation_report=self.context.invalidation_report,
                                              task_name=type(self).__name__,
                                              task_version=self.implementation_version_str(),
                                              artifact_write_callback=self.maybe_write_artifact)
 
+    # If this Task's execution has been forced, invalidate all our target fingerprints.
+    if self._cache_factory.ignore and not self._force_invalidated:
+      self.invalidate()
```

Comment:
Rather than being imperative, it seems like this could also be an argument to `cache_manager.check`, to cause the cache not to be used.

The difference in behaviour would essentially be that not _all_ targets for the task would be invalidated: only the ones actually in play within this context. So if you ran `./pants --cache-ignore test X` and then `./pants test Y`, you might still have some valid targets for Y if they weren't in X's transitive deps.